File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    answer = []
    dic = {}
    
    for row in keymap:
        for idx, char in enumerate(row):
            if char in dic:
                dic[char] = min(dic[char], idx + 1)
            else:
                dic[char] = idx + 1
                
    for target in targets:
        ans = 0
        for char in target:
            if char in dic:
                ans += dic[char]
            else:
                ans = -1
                break
        answer.append(ans)
        
    return answer

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.swift
import Foundation

func solution(_ keymap:[String], _ targets:[String]) -> [Int] {
    
    var answer = [Int]()
    
    var maps: [Character: Int] = [:]
    for key in keymap {
        let chars = Array(key)
        for (idx, c) in chars.enumerated() {
            if let val = maps[c] {
                maps[c] = min(val,idx+1) 
            } else {
                maps[c] = idx+1
            }
        }
    }
    
    for target in targets {
        let t = Array(target)
        var total = 0
        var canMap = true
        for x in t {
            if let cnt = maps[x] {
                total += cnt
            } else {
                canMap = false
                break
            }
        }
        answer.append(canMap ? total : -1)
    }
    
    return answer
}